### PR TITLE
De-duplicate getRandomBlock across the repo

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Closes: #1301 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

Refactor:
- Replaced local `getRandomBlock` and `getRandomHeader` functions with global `types.GetRandomBlock` and `types.GetRandomHeader` functions, improving code reusability and consistency across multiple files.

Tests:
- Updated test cases in `da_test.go`, `full_client_test.go`, `store_test.go`, and `block_test.go` to use the new global functions, ensuring tests remain accurate and reliable.

New Features:
- Enhanced `GetRandomBlock` and `GetRandomHeader` functions in `test_utils.go` to generate more comprehensive and realistic data, improving the quality of testing and debugging processes.

Style:
- Modified function signatures in `full_client_test.go` and `block_test.go` for better code readability and maintainability.

### CODE REVIEW

File: types/test_utils.go [hunk 1]
```diff
@@ -25,32 +25,71 @@ func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey)
...
-				App:   InitStateVersion.Consensus.App,
+func GetRandomBlock(height uint64, nTxs int) *Block {
+	block := &Block{
+		SignedHeader: SignedHeader{
+			Header: Header{
+				BaseHeader: BaseHeader{
+					Height: height,
+				},
+				Version:         Version{Block: InitStateVersion.Consensus.Block},
+				AggregatorsHash: make([]byte, 32),
```
Comment (MSevey) [inline types/test_utils.go:36]: Should this also use `GetRandomBytes`?

File: types/test_utils.go [hunk 2]
```diff
@@ -25,32 +25,71 @@ func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey)
...
+	}
+
+	return block
+}
+
+func GetRandomHeader() Header {
+	return Header{
+		BaseHeader: BaseHeader{
+			Height:  12,
+			Time:    uint64(time.Now().Local().Day()),
```
Comment (MSevey) [inline types/test_utils.go:66]: Do we know why this is `.Local().Day()`? This is setting the time to 1-31. 
Should this be `uint64(time.UnixNano())`?

File: types/test_utils.go [hunk 3]
```diff
@@ -25,32 +25,71 @@ func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey)
...
+			ChainID: "test",
```
Comment (MSevey) [inline types/test_utils.go:67]: we should set a const `TestChainID`

File: types/test_utils.go [hunk 4]
```diff
@@ -25,32 +25,71 @@ func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey)
...
+		},
+		Version: Version{
+			Block: 1,
+			App:   2,
```
Comment (MSevey) [inline types/test_utils.go:71]: We should set Block and App version constants

File: types/test_utils.go [hunk 5]
```diff
@@ -25,32 +25,71 @@ func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey)
...
+		block.Data.IntermediateStateRoots.RawRootsList = nil
+	}
+
+	return block
+}
+
+func GetRandomHeader() Header {
+	return Header{
+		BaseHeader: BaseHeader{
+			Height:  12,
```
Comment (MSevey) [inline types/test_utils.go:65]: Hardcoded height?

**Review by MSevey — CHANGES_REQUESTED** [review]
I think there is some more clean up we can do.

**Comment by MSevey** [comment]
If you want the codecov to pass you could have Co-Pilot write tests for the test_utils.go helpers that verifys multiple calls of the random methods actually produce random values. 

```
func Test() {
  for 0-10 {
    require.NotEqual(RandomFunc(), RandomFunc())
  }
}
```

File: types/test_utils.go (hunk 1 above)
Comment (Manav-Aggarwal) [reply types/test_utils.go:36]: Used `GetRandomHeader` instead

**Comment by Manav-Aggarwal** [comment]
> If you want the codecov to pass you could have Co-Pilot write tests for the test_utils.go helpers that verifys multiple calls of the random methods actually produce random values.
> 
> ```
> func Test() {
>   for 0-10 {
>     require.NotEqual(RandomFunc(), RandomFunc())
>   }
> }
> ```

Wrote some tests for the utils that look like this. For other methods, since the hash of a struct is used as a key in the map, specifically for methods, GetRandomBlock, and GetRandomSignedHeader, the hash ends up being the same as the header hash, so skipped those methods for now.

**Review by MSevey — APPROVED** [review]

**Review by nashqueue — APPROVED** [review]

**Pull request merged**